.github/workflows/security-scan.yml

```
@@ -0,0 +1,64 @@
+    permissions:
+      contents: read
+      security-events: write
+      pull-requests: write # Required for PR comments if enabled
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Starting a demo API to scan against (optional, for testing)
+      - name: Set up Python
+        uses: actions/setup-python@v4
```

actionlint (1.7.10)

[error] the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)